### PR TITLE
Fix duplicate field name HPC-5858, HPC-6341

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

**Comment by alvarosimon** [comment]
@itkovian we should change the ONE pattern tests as well `..data/opennebula`

**Comment by alvarosimon** [comment]
and sorry for this `level`  field name was a bad choice 😭

**Comment by itkovian** [comment]
Running on both tangelas. Works:

~~~~
[ageorges@tangela2 ~]$ curl -XGET '192.168.24.1:9200/logstash-2016.12.14/_search?q=onelevel:D' 2>/dev/null
{"took":6,"timed_out":false,"_shards":{"total":5,"successful":5,"failed":0},"hits":{"total":32,"max_score":0.13353139,"hits":[{"_index":"logstash-2016.12.14","_type":"%{@type}","_id":"AVj-RlcEwryRoSoLklk0","_score":0.13353139,"_source":{"message":"<31>1 2016-12-14T17:59:32.429895+01:00 one11 oned[18561]: - oned[18561]::  [Z0][VMM][D]: VM 58 successfully monitored: STATE=a CPU=0.5 MEMORY=4373832 NETRX=33780275293 NETTX=3305467381 DISK_SIZE=[ ID=52, SIZE=40960 ] DISK_SIZE=[ ID=1, SIZE=0 ]  ","@version":"1","@timestamp":"2016-12-14T16:59:32.429Z","host":"127.0.0.1","port":34389,"type":"syslog","syslog_abspri":"31","syslog_version":"1","program":"oned","pid":"18561","onelevel":"D","vmid":58,"vmstate":"a","vmucpu":0.5,"vmmem":4373832,"vmnetrx":33780275293,"vmnettx":3305467381,"received_at":"2016-12-14T16:59:32.430Z","received_from":"%{@source_host}","@source_host":"one11","@message":"%{syslog_message}"}},{"_index":"logstash-2016.12.14","_type":"%{@type}","_id":"AVj-TiI6wryRoSoLk9fJ","_score":0.13353139,"_source":{"message":"<31>1 2016-12-14T18:08:02.318861+01:00 one11 oned[18561]: - oned[18561]::  [Z0][VMM][D]: VM 54 successfully monitored: STATE=a CPU=6.0 MEMORY=8558204 NETRX=62610596383 NETTX=65953818296 DISK_SIZE=[ ID=38, SIZE=40960 ] DISK_SIZE=[ ID=1, SIZE=0 ]  ","@version":"1","@timestamp":"2016-12-14T17:08:02.318Z","host":"127.0.0.1","port":34389,"type":"syslog","syslog_abspri":"31","syslog_version":"1","program":"oned","pid":"18561","onelevel":"D","vmid":54,"vmstate":"a","vmucpu":6.0,"vmmem":8558204,"vmnetrx":62610596383,"vmnettx":65953818296,"received_at":"2016-12-14T17:08:02.319Z","received_from":"%{@source_host}","@source_host":"one11","@message":"%{syslog_message}"}},{"_index":"logstash-2016.12.14","_type":"%{@type}","_id":"AVj-Tl20wryRoSoLk92t","_score":0.13353139,"_source":{"message":"<31>1 2016-12-14T18:08:17.506301+01:00 one11 oned[18561]: - oned[18561]::  [Z0][VMM][D]: VM 53 successfully monitored: STATE=a CPU=7.0 MEMORY=8586248 NETRX=72742434908 NETTX=181140846794 DISK_SIZE=[ ID=50, SIZE=40960 ] DISK_SIZE=[ ID=1, SIZ
<snip>
~~~~

File: tests/data/opennebula [hunk 1]
```diff
@@ -2,7 +2,7 @@ data = [
 {
     "raw" : "2015-03-19T08:13:12.916717+01:00 one01 oned: [Z0][VMM][D]: VM 65 successfully monitored: STATE=a CPU=1.0 MEMORY=4381156 NETRX=131913740597 NETTX=25058980345 DISK_SIZE=[ ID=0, SIZE=10240 ] DISK_SIZE=[ ID=1, SIZE=0 ]",
     "expected" : {
-           "level" : "D",
+           "onelevel" : "D",
```
Comment (alvarosimon) [inline tests/data/opennebula]: You have to replace the lines below as well `s/level/onelevel/g`

Comment (itkovian) [inline tests/data/opennebula]: I could have sworn I hit `n` when searching for `level` :/

File: logstash/filters/bytes2human.rb [hunk 2]
```diff
@@ -42,7 +42,7 @@ def filter(event)
   def convert(event)
     @convert.each do |field, direction|
       next unless event.include?(field)
-      original = event[field]
+      original = event.get(field)
```
Comment (stdweird) [inline logstash/filters/bytes2human.rb]: might want to add this to upstream too https://github.com/robin13/logstash-filter-bytes2human

**Comment by stdweird** [comment]
jenkins needs new logstash

**Comment by stdweird** [comment]
@itkovian can you bump the logstash version in the Jenkinsfile?

**Comment by stdweird** [comment]
@itkovian the wget url of the jenkinsfile doesn't work...

File: logstash/filters/bytes2human.rb (hunk 2 above)
Comment (itkovian) [inline logstash/filters/bytes2human.rb]: Ah, we stole this? :-p

**Comment by itkovian** [comment]
Hmm, still something weird that should be fixed. I'll also drop the .Q for the longterm indices. joda-time refuses patches and I will not rebuild the complete thing time and again.

File: logstash/filters/bytes2human.rb (hunk 2 above)
Comment (itkovian) [inline logstash/filters/bytes2human.rb]: Upstream PR still open ...

**Comment by alvarosimon** [comment]
@itkovian jenkins is not happy, I'm not able to access to the failing tests... I have to ask to @JensTimmerman

**Comment by JensTimmerman** [comment]
@alvarosimon access granted

**Comment by alvarosimon** [comment]
@JensTimmerman mm I'm still getting the message `lsimngar is missing the Overall/Read permission`

**Comment by alvarosimon** [comment]
@JensTimmerman ah now it works thanks!

**Comment by itkovian** [comment]
Jenkins machines do not support java (1.)8. Oh well.

**Comment by JensTimmerman** [comment]
yes they do, you just have to configure it.

**Comment by JensTimmerman** [comment]
`Can't load line as json: LoadError: JRuby ext built for wrong Java version in `com.purbon.jrmonitor.JRMonitorService': java.lang.UnsupportedClassVersionError: com/purbon/jrmonitor/JRMonitorService : Unsupported major.minor version 52.0.`
so this seems to be a different error? Java is working...

File: Jenkinsfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env groovy
 
 def VIRTUALENV_VERSION = "15.0.3"
-def LOGSTASH_VERSION = "2.3.4"
+def LOGSTASH_VERSION = "5.1.1"
 
 node {
     stage 'Checkout'
```
Comment (JensTimmerman) [inline Jenkinsfile]: set
```
env.JAVA_HOME="${tool 'jdk-8u45'}"
env.PATH="${env.JAVA_HOME}/bin:${env.PATH}"
```

**Pull request closed without merging**